### PR TITLE
Remove the curses dialog, thereby deprecating the --help and --dialog command line options

### PR DESCRIPTION
**Pull request by ohemorange**
As per #3324.

### CODE REVIEW

**Comment by ohemorange** [comment]
Coverage went down because I removed covered lines, and also because I fixed some of the testing code that runs when the tests break.

File: certbot/cli.py [hunk 1]
```diff
@@ -677,7 +666,7 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
         "e.g. -vvv.")
     helpful.add(
         None, "-t", "--text", dest="text_mode", action="store_true",
-        help="Use the text output instead of the curses UI.")
+        help="Use the text output.")
```
Comment (jsha) [inline certbot/cli.py]: Shouldn't this also say "Deprecated - this has no effect?"

File: docs/contributing.rst [hunk 2]
```diff
@@ -246,7 +246,7 @@ configuration checkpoints and rollback.
 Display
 ~~~~~~~
 
-We currently offer a pythondialog and "text" mode for displays. Display
+We currently offer a "text" mode for displays. Display
```
Comment (jsha) [inline docs/contributing.rst]: This should probably say "We currently only offer.."

File: certbot/cli.py (hunk 1 above)
Comment (ohemorange) [inline certbot/cli.py]: That it should!

File: docs/contributing.rst (hunk 2 above)
Comment (ohemorange) [inline docs/contributing.rst]: Very true.

File: docs/cli-help.txt [hunk 3]
```diff
@@ -22,14 +22,11 @@ optional arguments:
   -v, --verbose         This flag can be used multiple times to incrementally
                         increase the verbosity of output, e.g. -vvv. (default:
                         -2)
-  -t, --text            Use the text output instead of the curses UI.
```
Comment (bmw) [inline docs/cli-help.txt]: This is another of our "only modify during releases" files, except this one doesn't have a test to catch the problem. The reason we don't modify this is we use this file to generate our documentation at https://certbot.eff.org/docs/using.html#command-line and we want to keep this information synced with the current release.

File: certbot/cli.py (hunk 1 above)
Comment (ohemorange) [inline certbot/cli.py]: Oh right now I remember, I left this in on the off chance we eventually implement something else for the `IDialog` interface. I'll make this clear though.

File: certbot/cli.py [hunk 4]
```diff
@@ -686,7 +675,7 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
               "which ones are required if it finds one missing")
     helpful.add(
         None, "--dialog", dest="dialog_mode", action="store_true",
-        help="Run using interactive dialog menus")
+        help="Deprecated - this has no effect")
```
Comment (bmw) [inline certbot/cli.py]: I think we should suppress help output here rather than continuing to print messages about a deprecated flag. We have a utility function for deprecating arguments called [add_deprecated_argument](https://github.com/certbot/certbot/blob/master/certbot/util.py#L369) that does this. I think the only reason we may not want to use this is it will cause a warning to be printed whenever the flag is used. This might be overkill for flags like `--text` that might be used in renewal scripts.

If we don't want to print a warning, we should use `argparse.SUPPRESS` here as it's done in `add_deprecated_argument`.

Comment (ohemorange) [inline certbot/cli.py]: I wonder if literally anyone ever used the `--dialog` flag, I have no problem deprecating that one properly.

As for `--text`, we have a whole bunch of tests that use it, I think suppressing it is the right move.

File: docs/cli-help.txt (hunk 3 above)
Comment (ohemorange) [inline docs/cli-help.txt]: Ok it seems the correct choice here is to not generate the documentation on that website based on master, but instead to generate it based on the release branch. Is there already an issue for that, or a reason why we don't already do that?

**Comment by ohemorange** [comment]
Re: `certbot/interfaces.py` - I'm fine keeping it in the interface and continuing to ignore it, on the off chance that this breaks some closed-source plugin.

File: docs/cli-help.txt (hunk 3 above)
Comment (bmw) [inline docs/cli-help.txt]: We often change the documentation for reasons other than a new release. We could delay publishing changes until a release has been done, but blocking all improvements to our documentation to keep this file in lockstep seems undesirable to me.

File: certbot/cli.py (hunk 4 above)
Comment (bmw) [inline certbot/cli.py]: I agree. Let's deprecate `--dialog` and suppress help output from `--text`.

**Comment by bmw** [comment]
I personally don't think reverting changes to `certbot/interfaces.py` and the `notification` methods in `certbot/display/util.py` is necessary. I was just bringing that up as an example of the kind of thing we need be careful about. I looked into it though and it doesn't seem to be a problem.

While it's possible there is some closed source plugin out there, none of us have ever heard about it. I personally think we should take advantage of this and use it as a chance to clean up the API a bit. I don't have a strong opinion here though, so feel free to revert these changes if you wish.

File: docs/cli-help.txt (hunk 3 above)
Comment (ohemorange) [inline docs/cli-help.txt]: Are these documentation changes urgent enough that they can't wait for a new release? If we release regularly, this shouldn't be a problem.

Comment (bmw) [inline docs/cli-help.txt]: They are rarely urgent, however, they do offer a better experience for the user. Why should we tie our hands and force ourselves to sit on these changes until we do a release?

And while I wouldn't call them urgent, there are definitely times we want to make changes outside of a release. For example, the stuff that I've been working on involves doc changes and needs to go live almost immediately so it is up before super secret project that I'm not sure I'm allowed to talk about publicly involving encryption, pipes, and Noah the hacker is released. Another example of changes like this is changes in OS packaging such as when a new version of a distro is released.

Comment (bmw) [inline docs/cli-help.txt]: Yeah I think there are a number of things we could do to make this a little nicer for developers. I think it could get a bit tricky though.

And no need to make a separate PR. These changes happen automatically in our [release script](https://github.com/certbot/certbot/blob/master/tools/release.sh#L149).

Comment (ohemorange) [inline docs/cli-help.txt]: Cool! I also modified `docs/contributing.rst`, should I make a separate PR for that one?

Comment (bmw) [inline docs/cli-help.txt]: Oh sorry. I missed that one. Yeah I suppose you should make a separate PR that we can merge during the release.

**Comment by ohemorange** [comment]
All requested changes made!
